File: app/src/main/java/org/test/koltintest/UsersEditFragment.java
package org.test.koltintest;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.TextInputLayout;
import android.support.v4.app.Fragment;
import android.text.Editable;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import org.test.koltintest.model.User;

import java.util.Objects;

public class UsersEditFragment extends Fragment {
    private static final String ARG_USER = "user";

    public static UsersEditFragment create() {
        return create(null);
    }

    public static UsersEditFragment create(@Nullable User user) {
        Bundle args = new Bundle();
        args.putParcelable(ARG_USER, user);

        UsersEditFragment fragment = new UsersEditFragment();
        fragment.setArguments(args);
        return fragment;
    }

    {
        setHasOptionsMenu(true);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_users_edit, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        if (savedInstanceState == null) {
            User user = getArguments().getParcelable(ARG_USER);

            if (user != null) {
                TextInputLayout name = view.findViewById(R.id.name);
                TextInputLayout profession = view.findViewById(R.id.profession);
                TextInputLayout age = view.findViewById(R.id.age);

                name.getEditText().setText(user.getName());
                profession.getEditText().setText(user.getProfession());
                age.getEditText().setText(Objects.toString(user.getAge(), null));
            }
        }
    }

    @Override
    public void onCreateOptionsMenu(@NonNull Menu menu, @NonNull MenuInflater inflater) {
        super.onCreateOptionsMenu(menu, inflater);

        inflater.inflate(R.menu.fragment_users_edit, menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()) {
            case R.id.save:
                if (validate()) {
                    // TODO perform save here

                    getFragmentManager().popBackStack();
                }
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

    private boolean validate() {
        View view = getView();
        TextInputLayout name = view.findViewById(R.id.name);
        TextInputLayout profession = view.findViewById(R.id.profession);
        TextInputLayout age = view.findViewById(R.id.age);

        boolean result = validate(name, true);
        result &= validate(profession, result);
        // result &= validate(age, result);
        return result;
    }

    private boolean validate(TextInputLayout layout, boolean setFocus) {
        Editable value = layout.getEditText().getText();

        if (TextUtils.isEmpty(value)) {
            layout.setError(getText(R.string.err_field_required));

            if (setFocus) {
                layout.requestFocus();
            }
            return false;

        } else {
            layout.setErrorEnabled(false);
        }
        return true;
    }

}

File: app/src/main/java/org/test/koltintest/UsersEditFragment.kt
package org.test.koltintest

import android.os.Bundle
import android.view.*
import androidx.core.os.bundleOf
import androidx.fragment.app.Fragment
import kotlinx.android.synthetic.main.fragment_users_edit.*
import org.test.koltintest.model.User
import org.test.text
import org.test.validate

class UsersEditFragment : Fragment() {

    private val user by lazy { arguments?.getParcelable(ARG_USER) as User? }

    init {
        setHasOptionsMenu(true)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?) =
            inflater.inflate(R.layout.fragment_users_edit, container, false)!!

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        if (savedInstanceState == null) {
            name.text = user?.name
            profession.text = user?.profession
            age.text = user?.age?.toString()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        super.onCreateOptionsMenu(menu, inflater)

        inflater.inflate(R.menu.fragment_users_edit, menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.save -> {
                if (validate()) {
                    // TODO perform save here

                    fragmentManager!!.popBackStack()
                }
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }

    private fun validate() = arrayOf(name, profession)
            .fold(true) { result, layout -> result and layout.validate(result) }

    companion object {
        private const val ARG_USER = "user"

        fun create(user: User? = null) = UsersEditFragment().apply {
            arguments = bundleOf(ARG_USER to user)
        }
    }

}
